Docs offcanvas navbar

mdo opened the pull request:
<img width="1792" alt="Screen Shot 2021-07-12 at 1 38 19 PM" src="https://user-images.githubusercontent.com/98681/125352715-795f3380-e316-11eb-8797-150c9671eefd.png">

Work in progress to update our docs navbar for v5.2.0 to use the offcanvas navbar styles.

- [ ] Verify padding changes to layout (set new variable perhaps and use that?)
- [ ] Should docs sidebar be an offcanvas too?

Preview: https://deploy-preview-34487--twbs-bootstrap.netlify.app/

mdo committed:
Redesign docs navbar to use offcanvas, tweak overall padding while I'm here

XhmikosR commented on site/layouts/partials/docs-navbar.html:
```diff
@@ -4,62 +4,72 @@
-      </ul>
+    <div class="offcanvas offcanvas-end" tabindex="-1" id="bdNavbar" aria-labelledby="bdNavbarOffcanvasLabel">
+      <div class="offcanvas-header pb-0">
+        <h5 class="offcanvas-title text-white" id="bdNavbarOffcanvasLabel">Bootstrap</h5>
```
Just thinking out loud, might look better if it's aligned center?

XhmikosR commented on site/assets/scss/_component-examples.scss:
```diff
@@ -302,7 +302,7 @@
 
 .highlight {
-  padding: 1rem;
+  padding: var(--bs-gutter-x);
```
Are you sure this is related? And some of the other changes in the CSS.

mdo replied:
Yeah, these and other changes need another pass, but they're intentional. Gives the content a little more breathing room. Problem is I need to be careful about where I use `var()` vs fixed values because our examples would accidentally inherit those changed values.

mdo committed:
Match code snippets to new gutter width

ffoodd reviewed (approved):
LGTM!

We seem to use `1.25rem` a lot indeed, setting a new variable might help a bit.

Regarding docs sidebar, I don't think it should be using offcanvas. Having two different navs help in identifying them , IMHO. But that's more a UX decision, my opinion might not have much value :D 

The PR is looking fine as-is, and is working fine AFAIC.

mdo committed:
Replace value with docs-specific variable

Opted not to do a CSS variable here since on .bd-content it would be inherited into our examples

ffoodd reviewed (approved):
Nice 👌

XhmikosR committed:
Merge branch 'main' into docs-offcanvas-navbar
